src/Common/src/Interop/Windows/NtDll/Interop.RtlGetVersion.cs

```diff
@@ -11,21 +13,57 @@ internal partial class NtDll
         [DllImport(Libraries.NtDll, ExactSpelling=true)]
         private static extern int RtlGetVersion(ref RTL_OSVERSIONINFOEX lpVersionInformation);
 
-        internal static unsafe string RtlGetVersion()
+        private static unsafe Tuple<uint, uint, uint, string> RtlGetVersionInternal()
         {
             var osvi = new RTL_OSVERSIONINFOEX();
             osvi.dwOSVersionInfoSize = (uint)sizeof(RTL_OSVERSIONINFOEX);
-            const string version = "Microsoft Windows";
+            
             if (RtlGetVersion(ref osvi) == 0)
             {
                 return osvi.szCSDVersion[0] != '\0' ?
-                    string.Format("{0} {1}.{2}.{3} {4}", version, osvi.dwMajorVersion, osvi.dwMinorVersion, osvi.dwBuildNumber, new string(&(osvi.szCSDVersion[0]))) :
-                    string.Format("{0} {1}.{2}.{3}", version, osvi.dwMajorVersion, osvi.dwMinorVersion, osvi.dwBuildNumber);
+                    new Tuple<uint, uint, uint, string>(osvi.dwMajorVersion, osvi.dwMinorVersion, osvi.dwBuildNumber, new string(&(osvi.szCSDVersion[0]))) :
+                    new Tuple<uint, uint, uint, string>(osvi.dwMajorVersion, osvi.dwMinorVersion, osvi.dwBuildNumber, null);
             }
             else
             {
-                return version;
+                return null;
+            }
+        } 
+
+        internal static string RtlGetVersion()
+        {
+            var sb = new StringBuilder();
+            const string version = "Microsoft Windows";
+            sb.Append(version);
+            Tuple<uint, uint, uint, string> info = RtlGetVersionInternal();
+            if (info != null)
+            {
+                sb.AppendFormat(" {0}.{1}.{2}", info.Item1, info.Item2, info.Item3);
+                if (info.Item4 != null)
+                {
+                    sb.AppendFormat(" {0}", info.Item4);
+                }
             }
+
+            return sb.ToString();
+        }
```

Comment:
Why is the tuple needed rather than just calling RtlGetVersion in each of these helpers?  The Tuple doesn't really seem to simplify anything, adds an allocation (though that's not a big deal in this context), and actually introduces a potential null reference exception below in the case where RtlGetVersion fails (as written, RtlGetVersionInternal will return null).  Plus, this StringBuilder-based implementation is more verbose and likely more expensive than the previous code (though, again, in the context it's used, the extra overhead likely doesn't really matter).

Reply:
Thanks for the comment. I've reverted it back and added a method with `out` param. Initially I thought about some encapsulation for `NtDll` when saw `extern` is `private` but i image it is not handy to do this for Interop.